rust/libnewsboat/src/configcontainer.rs

```diff
@@ -0,0 +1,1360 @@
+            ConfigData::new(&gettext("%N %V - %?F?Feeds&Your feeds? (%u unread, %t total)%?F? matching filter '%F'&?%?T? - tag '%T'&?"), ConfigDataType::Str));
+
+        config_data.insert(
+            "filebrowser-title-format".to_string(),
+            ConfigData::new(
+                &gettext("%N %V - %?O?Open File&Save File? - %f"),
+                ConfigDataType::Str,
+            ),
+        );
+
+        config_data.insert(
+            "dirbrowser-title-format".to_string(),
+            ConfigData::new(
+                &gettext("%N %V - %?O?Open Directory&Save File? - %f"),
+                ConfigDataType::Str,
+            ),
+        );
+
+        config_data.insert(
+            "help-title-format".to_string(),
+            ConfigData::new(&gettext("%N %V - Help"), ConfigDataType::Str),
+        );
+
+        config_data.insert(
+            "itemview-title-format".to_string(),
+            ConfigData::new(
+                &gettext("%N %V - Article '%T' (%u unread, %t total)"),
+                ConfigDataType::Str,
+            ),
+        );
+
+        config_data.insert(
+            "searchresult-title-format".to_string(),
+            ConfigData::new(
+                &gettext(
+                    "%N %V - Search results for '%s' (%u unread, %t total)%?F? matching filter '%F'&?",
+                ),
+                ConfigDataType::Str,
+            ),
+        );
+
+        config_data.insert(
+            "selectfilter-title-format".to_string(),
+            ConfigData::new(&gettext("%N %V - Select Filter"), ConfigDataType::Str),
+        );
+
+        config_data.insert(
+            "selecttag-title-format".to_string(),
+            ConfigData::new(&gettext("%N %V - Select Tag"), ConfigDataType::Str),
+        );
+
+        config_data.insert(
+            "urlview-title-format".to_string(),
+            ConfigData::new(&gettext("%N %V - URLs"), ConfigDataType::Str),
+        );
+
+        config_data.insert(
+            "wrap-scroll".to_string(),
+            ConfigData::new("no", ConfigDataType::Bool),
+        );
+
+        ConfigContainer {
+            config_data: Arc::new(Mutex::new(config_data)),
+        }
+    }
+
+    pub fn handle_action(&self, action: &str, params: &[String]) -> Result<(), ConfigHandlerError> {
+        let mut data = self.config_data.lock().unwrap();
+        if let Some(entry) = data.get_mut(action) {
+            if params.is_empty() {
+                return Err(ConfigHandlerError::TooFewParams);
+            }
+
+            if !entry.multi_option && params.len() > 1 {
+                return Err(ConfigHandlerError::TooManyParams);
+            }
+
+            let value = if entry.multi_option {
+                params.join(" ")
+            } else {
+                params[0].clone()
+            };
+
+            match entry.set_value(value) {
+                Ok(_) => Ok(()),
+                Err(msg) => Err(ConfigHandlerError::InvalidParams(msg)),
+            }
```

Comment:
Nit: idiomatic Rust for this would be `entry.set_value(value).map_err(ConfigHandlerError::InvalidParams)`.